src/genericsite/models.py

```diff
@@ -80,6 +82,136 @@ def For(cls, site: T.Union[Site, int]) -> models.Manager:
         return cls.objects.filter(site_id=site_id)
 
 
+######################################################################################
+# Media Objects
+######################################################################################
+class MediaObject(models.Model):
+    """Abstract base class for stored media files (image, audio, video, attachment).
+
+    Every media object has an `image_file` attribute so it can have a thumbnail of some
+    kind to display in visual contexts. Thumbnails of standard sizes can be accessed
+    via this field. We assume images will use this field as the main `content_field`.
+    Other subclasses should use this to store a thumbnail or cover art, and add a new
+    field to store their content file, and set the `content_field` attribute to the
+    name of the new field where their audio/video/attachment is stored.
+
+    If `mime_type` is not populated, it will be guessed when the instance is saved,
+    based on the file name stored in the `content_field` (using `mimetypes.guess_type`).
+    """
+
+    class Meta:
+        abstract = True
+
+    content_field = "image_file"
```

Comment:
Since all media objects are going to going to end up with an image here, do you think alt_text should be here instead of on the Image model?

Reply:
Excellent point. I'm going to change this.